tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs

```diff
@@ -66,6 +69,20 @@ internal ILocalRepositoryProvider GetLocalProvider(string rootPath)
         Log.Information("GetLocalProvider succeeded");
         return provider;
     }
+
+    IDictionary<string, object> IExtraFolderPropertiesHandler.GetProperties(string[] propertyStrings, string rootFolderPath, string relativePath)
+    {
+        var localProvider = GetLocalProvider(rootFolderPath);
+        var localProviderResult = localProvider.GetRepository(rootFolderPath);
+        if (localProviderResult.Result.Status == ProviderOperationStatus.Failure)
+        {
+            log.Information("Could not open local repository.");
+            log.Information(localProviderResult.Result.DisplayMessage);
+            throw new ArgumentOutOfRangeException(nameof(rootFolderPath), rootFolderPath, localProviderResult.Result.DisplayMessage);

```

Comment:
Dev Home will not be the receiver of this exception but does it need to surface any error to the user? Thanks!

Reply:
This will get called multiple times per file and would result in a bit of spamminess. I'm assuming that if we are failing to open the repository here, we'd fail to open the repository in the management or settings screen.

When we add telemetry, I'd like to add some to this case so we can track if this is something that users' machines hit. If so, we could figure out how and when to surface it vs log something for later. Might be something good to think about as we evolve the repo management user experience flows.